docs/api/v3.rst

```diff
@@ -0,0 +1,494 @@
+API v3
+======
+
+The Read the Docs API uses :abbr:`REST (Representational State Transfer)`.
+JSON is returned by all API responses including errors
+and HTTP response status codes are to designate success and failure.
+
+
+Authentication and authorization
+--------------------------------
+
+Requests to the Read the Docs public API are for public and private information
+and do require authentication even for requesting public information.
+
+Scopes
+~~~~~~
+
+The API can be accessed by using the HTTP header ``Authorization``
+with a specific ``Token`` that provide access to different resources.
+
+Available scopes
+++++++++++++++++
+
+* ``project``
+  * ``project:admin``: access to change admin settings (similar to Admin tab)
+  * ``project:read``: access to read all the information about the projects
+  * ``project:write``: access to anything related to perform write actions (triggering a Build for this project, activate/deactivate versions, etc)
+  * ``project:docs``: access any file from any version of the project's documentation
+
+
+.. TODO:
+
+   We could extend the scopes to be per-project or per-user. I'm not
+   sure if this is possible yet, but we should consider it.
+
+   * per-project: the user could go to the Project Admin tab and create a Token for this specific project
+   * per-user: the user could go to their own User settings and create token for all his projects
+
+
+Resources
+---------
+
+Projects
+~~~~~~~~
+
+Project list
+++++++++++++
+
+.. http:get::  /api/v3/project/
+
+    Retrieve a list of all the projects of the current logged in user.
+
+    **Example request**:
+
+    .. sourcecode:: bash
+
+        $ curl https://readthedocs.org/api/v3/project/
+
+    **Example response**:
+
+    .. sourcecode:: js
+
+        {
+            "count": 25,
+            "next": "/api/v3/project/limit=10&offset=10",
+            "previous": null,
+            "results": [PROJECTS]
+        }
+
+    :>json integer count: Total number of Projects.
+    :>json string next: URI for next set of Projects.
+    :>json string previous: URI for previous set of Projects.
+    :>json array results: Array of ``Project`` objects.
+
+.. TODO:
+
+   Add query string filters to narrow the query:
+     * privacy level
+     * language
+     * programming language
+     * repo url
+     * repo type
+     * version active
+     * version built
+     * all database?
+
+
+Project details
++++++++++++++++
+
+.. http:get::  /api/v3/project/(string:slug)/
```

Comment:
Everything should be accessed by `string:slug` or `id:int`.

https://geemus.gitbooks.io/http-api-design/content/en/requests/support-non-id-dereferencing-for-convenience.html

Reply:
I don't think we need to expose the ID.

Reply:
👍 to not exposing ID. `slug` is the public facing unique identifier in my mind. Though it is possible that it could change over time (by an admin), but I think that's an edge case.